pom.xml

```diff
@@ -38,22 +38,23 @@
 
   <name>Kubernetes Credentials Provider</name>
   <description>Provides a credentials store backed by Kubernetes.</description>
-  <url>https://jenkinsci.github.io/kubernetes-credentials-provider-plugin/</url>
+  <url>https://github.com/${gitHubRepo}</url>
```

Comment:
IIRC this is used in the UC and provided to the user in Jenkins' plugin management?

As such this should point to the actual plugins doc / homepage and not the repo (which is more for developers?) - so the current URL would seem to be more appropriate?

otherwise this will I guess somehow fix plugins,jenkins.io, but beak the plugin in Jenkins which I value more.

Reply:
I don't see this description anywhere in the plugins management page in Jenkins (I could be wrong though)

<img width="1270" alt="image" src="https://github.com/jenkinsci/kubernetes-credentials-provider-plugin/assets/91831478/bd124876-6bdf-450d-9f95-ed22a9fd8ab6">

Only the short description from https://github.com/jenkinsci/kubernetes-credentials-provider-plugin/blob/master/src/main/resources/index.jelly#L3 is displayed in Jenkins AFAIK.

Keeping https://jenkinsci.github.io/kubernetes-credentials-provider-plugin/ or https://github.com/jenkinsci/kubernetes-credentials-provider-plugin/blob/master/docs/README.md has the issues previously highlighted.

Having the repo URL here is how most plugins are set, and that's how the plugin site fetches the README from the repo to be displayed as description on plugins.jenkins.io.

Reply:
not the description the link when you click on the plugins name.

> Having the repo URL here is how most plugins are set, and that's how the plugin site fetches the README from the repo to be displayed as description on plugins.jenkins.io.

😢

Reply:
> not the description the link when you click on the plugins name.

The link from the plugin name in Jenkins plugin manager points to the plugin site, ex: https://plugins.jenkins.io/kubernetes-credentials-provider

<img width="490" alt="image" src="https://github.com/jenkinsci/kubernetes-credentials-provider-plugin/assets/91831478/1b513d08-9550-4b66-b9f1-f4fb0eae0e79">

Maybe I'm missing something here, but I didn't find any other link pointing to the URL defined in the pom.xml.

There is an alternative way to keep this URL as it was, but it's not pretty: it's possible to add an override in https://github.com/jenkins-infra/update-center2/blob/master/resources/wiki-overrides.properties

Reply:
<img width="905" alt="image" src="https://github.com/jenkinsci/kubernetes-credentials-provider-plugin/assets/494726/48b20005-d2c2-49ea-9e06-ef6d6891cb27">

Reply:
well this is interesting...

The CloudBees update site honours the URL given by the plugin, the OSS site does not..

```
curl https://updates.jenkins.io/current/update-center.json | tail --lines=+2 | head --lines=-1 | jq .plugins.'"kubernetes-credent
ials-provider"'
```

vs

```
 curl --location "https://jenkins-updates.cloudbees.com/update-center/envelope-core-mm/update-center.json?version=2.414.3.6"  | tail --lines=+2 | head --lines=-1 | jq .plugins.'"kubernetes-credentials-provider"'
 ```

Reply:
On another hand, the current plugin site page is not broken and could suffice, rendering this PR obsolete.

https://plugins.jenkins.io/kubernetes-credentials-provider/

<img width="880" alt="image" src="https://github.com/jenkinsci/kubernetes-credentials-provider-plugin/assets/91831478/0eae6e98-afaf-4245-8a25-94da94cfc9b1">

Reply:
I think all the changes here apart from the POM are good.  
The extra info in the README means you get told what the plugin does without having to jump to another page.

Reply:
I've restored the pom.xml content, and applied your suggestion, should be good for a final review now.